### PR TITLE
Feat: Add remove all selections option for ui.text_annotator

### LINKED ISSUE #1470: Remove All option in text annotator
#### Is your feature request related to a problem? Please describe
Sometimes there is a need to clear all annotations of an entity and start again.

#### Describe the solution you'd like
An option in Text Annotator to `Remove All` similar to the Image Annotator.

### PR DESCRIPTION
**Pull request by marek-mihok**
https://user-images.githubusercontent.com/23740173/181818139-0723c595-06db-44cb-a049-de7e2fd870a6.mov

Closes #1470

### CODE REVIEW

File: ui/src/text_annotator.tsx [hunk 1]
```diff
@@ -113,7 +113,15 @@ const
...
       background: cssVar('$card'),
     },
-    readonly: { pointerEvents: 'none' }
+    readonly: { pointerEvents: 'none' },
+    checkbox: {
+      $nest: {
+        '&:hover': {
+          backgroundColor: cssVar('$neutralLighter'),
+          color: cssVar('$neutralDark')
+        }
```
Comment (mturoci) [inline ui/src/text_annotator.tsx:122]: I assume this is for consistency with `Remove all`, right?

File: ui/src/text_annotator.tsx [hunk 2]
```diff
@@ -202,6 +210,11 @@ export
         setTokens([...tokens])
         submitWaveArgs()
       },
+      removeAllAnnotations = () => {
+        for (let idx = 0; idx < tokens.length; idx++) tokens[idx].tag = undefined
```
Comment (mturoci) [inline ui/src/text_annotator.tsx:214]: Let's use `forEach` if possible.

File: ui/src/text_annotator.tsx [hunk 3]
```diff
@@ -263,7 +276,7 @@ export
             className={clas(css.mark, isFirst ? css.firstMark : '', isLast ? css.lastMark : '')}
             style={{ backgroundColor: cssVar(color), color: getContrast(color) }}>
             {text}
-            <Fluent.Icon iconName='CircleAdditionSolid' styles={{ root: removeIconStyle }} className={clas(css.removeIcon, 'wave-w6')} onMouseUp={removeAnnotation(idx)} />
+            <Fluent.Icon iconName='CircleAdditionSolid' data-test={'remove-icon'} styles={{ root: removeIconStyle }} className={clas(css.removeIcon, 'wave-w6')} onMouseUp={removeAnnotation(idx)} />
```
Comment (mturoci) [inline ui/src/text_annotator.tsx:279]: Nit
```suggestion
            <Fluent.Icon iconName='CircleAdditionSolid' data-test='remove-icon' styles={{ root: removeIconStyle }} className={clas(css.removeIcon, 'wave-w6')} onMouseUp={removeAnnotation(idx)} />
```

**Review by mturoci — CHANGES_REQUESTED** [review]
Thanks @marek-mihok! A few comments from me.

File: ui/src/text_annotator.tsx (hunk 1 above)
Comment (marek-mihok) [reply ui/src/text_annotator.tsx:122]: Yes, you're right!

**Review by mturoci — APPROVED** [review]

**Pull request merged**